state-chain/primitives/src/lib.rs

```diff
@@ -50,6 +50,15 @@ pub type PrewitnessedDepositId = u64;
 
 pub type BoostPoolTier = u16;
 
+// TODO: Consider increasing Price to U512 or switch to a f64 (f64 would only be for the external
+// price representation), as at low ticks the precision in the price is VERY LOW, but this does not
+// cause any problems for the AMM code in terms of correctness
+/// This is the ratio of equivalently valued amounts of asset One and asset Zero. The price is
+/// always measured in amount of asset One per unit of asset Zero. Therefore as asset zero becomes
+/// more valuable relative to asset one the price's literal value goes up, and vice versa. This
+/// ratio is represented as a fixed point number with `PRICE_FRACTIONAL_BITS` fractional bits.
+pub type Price = U256;
```

Comment:
It looks like this is only used in the broker api? As per my other comment - maybe we can expose something friendlier.

Reply:
Sounds like the product prefers using Price directly as Price/U256. I think the ability to represent a fractional price with a fixed point number is a very useful abstraction in general. I could keep the alias private, then expose U256 and then make a comment somewhere to say that "actually it is represented as `Price` from the amm module", but it is certainly not better. Maybe we could make it a proper type like you are suggesting, but that sounds like it can be its own (not a high priority) PR.

Reply:
I still think that exposing the internal type directly is the wrong decision but ok.